vignettes/independent_filtering_plots.Rmd

```diff
@@ -0,0 +1,219 @@
+  } else if(outputFormat == 'html') {
+    ret <- paste("<font color='",color,"'>",x,"</font>",sep="")
+  } else {
+    ret <- x
+  }
+  return(ret)
+}
+```
+
+```{r setup, echo=FALSE}
+options(width=80 )
+```
+
+# Introduction
+
+This vignette illustrates use of some functions in the *genefilter*
+package that provide useful diagnostics for independent
+filtering [@BourgonIndependentFiltering]:
+
+-   `kappa_p` and `kappa_t`
+
+-   `filtered_p` and `filtered_R`
+
+-   `filter_volcano`
+
+-   `rejection_plot`
+
+# Data preparation
+
+Load the ALL data set and the *genefilter* package:
+
+```{r libraries, message=FALSE, warning=FALSE}
+library("genefilter")
+library("ALL")
+data("ALL")
+```
+
+
+Reduce to just two conditions, then take a small subset of arrays from
+these, with 3 arrays per condition:
+
+```{r sample_data, cache=TRUE}
+bcell <- grep("^B", as.character(ALL$BT))
+moltyp <- which(as.character(ALL$mol.biol) %in%
+c("NEG", "BCR/ABL"))
+ALL_bcrneg <- ALL[, intersect(bcell, moltyp)]
+ALL_bcrneg$mol.biol <- factor(ALL_bcrneg$mol.biol)
+n1 <- n2 <- 3
+set.seed(1969)
+use <- unlist(tapply(1:ncol(ALL_bcrneg),
+ALL_bcrneg$mol.biol, sample, n1))
+subsample <- ALL_bcrneg[,use]
+```
+
+We now use functions from *genefilter* to compute overall standard
+devation filter statistics as well as standard two-sample $t$ and
+releated statistics.
+
+```{r stats, cache=TRUE}
+S <- rowSds( exprs( subsample ) )
+temp <- rowttests( subsample, subsample$mol.biol )
+d <- temp$dm
+p <- temp$p.value
+t <- temp$statistic
+```
+
+
+# Filtering volcano plot
+
+Filtering on overall standard deviation and then using a standard
+$t$-statistic induces a lower bound of fold change, albeit one which
+varies somewhat with the significance of the $t$-statistic. The
+`filter_volcano` function allows you to visualize this effect.
+
+The output is shown in the left panel of Fig.^[1](#fig:volcano){reference-type="ref" reference="fig:volcano"}.
+
+The `kappa_p` and `kappa_t` functions, used to make the volcano plot,
+compute the fold change bound multiplier as a function of either a
+$t$-test $p$-value or the $t$-statistic itself. The actual induced bound
+on the fold change is $\kappa$ times the filter's cutoff on the overall
+standard deviation. Note that fold change bounds for values of $|T|$
+which are close to 0 are not of practical interest because we will not
+reject the null hypothesis with test statistics in this range.
+
+Table:\#tab:label
+```{r filter_volcano, fig.cap = "Wide figure. A plot produced by a code chunk with option `fig.wide = TRUE`.", echo=FALSE}
+par(mfrow=c(1, 2))
+
+S_cutoff <- quantile(S, .50)
+ss <- filter_volcano(d, p, S, n1, n2, alpha=.01, S_cutoff)
+
+t <- seq(0, 5, length=100)
+plot(t, kappa_t(t, n1, n2) * S_cutoff,
+     xlab="|T|", ylab="Fold change bound", type="l")
+```
+__Figure 1:__ __`r colFmt("Left panel: plot produced by the ", 'blue')`__ `filter_volcano` __`r colFmt("function ", 'blue')`__.\
```

Comment:
I think we don't have to worry so much about the coloring and use the caption in line 108.